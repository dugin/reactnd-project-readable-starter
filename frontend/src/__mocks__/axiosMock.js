import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import {categories, posts} from "./responseMock";

const mock = new MockAdapter(axios);

const URL = 'http://localhost:3001';

mock.onGet(URL + '/categories').reply(200, categories);

mock.onGet(URL + '/posts').reply(200, posts);
