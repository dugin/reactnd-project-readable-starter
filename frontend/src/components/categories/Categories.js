import React, {Component} from 'react';
import * as readableAPI from '../../api/readableAPI';
import './Categories.css';

class Categories extends Component {

    state = {categories: []};

    componentDidMount() {
        this.getAllCategories();
    }

    getAllCategories = () => {
        readableAPI.getAllCategories()
            .then(resp => {
                if (resp.status === 200)
                    this.setState({categories: resp.data.categories});

                else
                    throw new Error({status: resp.status, msg: resp.statusText})
            })
            .catch(err => console.error(err));
    };

    render() {
        return (
            <div className="Categories">
                {this.state.categories.map(c => (
                    <a href="#" key={c.name} className="badge badge-pill badge-secondary">{c.name}</a>
                ))}

            </div>
        )
    }
}

export default Categories;