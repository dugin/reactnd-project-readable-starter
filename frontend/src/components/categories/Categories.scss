.Categories {
  display: flex;
  justify-content: space-between;
  overflow-x: scroll;
  overflow-y: hidden;
 // padding: 0 15px;
  width: 100%;
 // position: absolute;
  left: 0;
  right: 0;
  top: 55px;

  .badge {
    padding: 10px;
    margin: 10px 5px 10px 0;
    font-size: 14px;
    min-width: 70px;
  }
}