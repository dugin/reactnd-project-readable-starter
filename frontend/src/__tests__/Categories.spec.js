import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Categories from "../components/categories/Categories";
import '../__mocks__/axiosMock';

describe('Categories Component', () => {

    const categories = shallow(<Categories/>);

    it('renders properly', () => {

        expect(toJson(categories)).toMatchSnapshot();

    });
});