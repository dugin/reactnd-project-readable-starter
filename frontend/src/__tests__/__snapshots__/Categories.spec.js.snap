// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Component renders properly 1`] = `
<div
  className="Categories"
/>
`;
